README.md

```diff
@@ -79,6 +79,247 @@ For `testStart` and `testEnd`, the corresponding test object is passed to the re
+  tests: [Test('test2')],
+  childSuites: [],
+  status: undefined,
+  testCounts: {
+    passed: undefined,
+    failed: undefined,
+    skipped: undefined,
+    total: 1
+  },
+  runtime: undefined
+```
+
+- **testStart**
+
+```js
+Test {
+  name: 'test2',
+  suiteName: 'suite2',
+  fullName: ['suite1', 'suite2', 'test2'],
+  status: undefined,
+  runtime: undefined,
+  errors: undefined,
+  assertions: undefined
+```
+
+- **testEnd**:
+
+```js
+Test {
+  name: 'test2',
+  suiteName: 'suite2',
+  fullName: ['suite1', 'suite2', 'test2'],
+  status: `passed`,
+  runtime: 1,
+  errors: [],
+  assertions: [{
+    passed: true,
+    actual: true,
+    expected: true,
+    message: `some message`,
+    stack: undefined
+  }]
+```
+
+- **suiteEnd**
+
+```js
+Suite {
+  name: 'suite2',
+  fullName: ['suite1', 'suite2'],
+  tests: [Test('test2')],
+  childSuites: [],
+  status: `passed`,
+  testCounts: {
+    passed: 1,
+    failed: 0,
+    skipped: 0,
+    total: 1
+  },
+  runtime: 1
+```
+
+- **testStart**
+
+```js
+Test {
+  name: 'test1',
+  suiteName: 'suite1',
+  fullName: ['suite1', 'test'],
+  status: undefined,
+  runtime: undefined,
+  errors: undefined,
+  assertions: undefined
+```
+
+- **testEnd**:
+
+```js
+Test {
+  name: 'test1',
+  suiteName: 'suite1',
+  fullName: ['suite1', 'test'],
+  status: `failed`,
+  runtime: 2,
+  errors: [{
+    passed: false,
+    actual: true,
+    expected: false,
+    message: `some message`,
+    stack: `stack trace`
+  }],
+  assertions: [{
+    passed: true,
+    actual: true,
+    expected: true,
+    message: `some message`,
+    stack: undefined
+  }, {
+    passed: false,
+    actual: true,
+    expected: false,
+    message: `some message`,
+    stack: `stack trace`
+  }]
+```
+
+- **suiteEnd**
+
+```js
+Suite {
+  name: 'suite1',
+  fullName: ['suite1'],
+  tests: [Test('test1')],
+  childSuites: [],
+  status: `failed`,
+  testCounts: {
+    passed: 1,
+    failed: 1,
+    skipped: 0,
+    total: 2
+  },
+  runtime: 3
+```
+
+- **runEnd**:
+
+```js
+Suite {
+  name: undefined,
+  fullName: [],
+  tests: [],
+  childSuites: [Suite('suite1')],
+  status: `failed`,
+  testCounts: {
+    passed: 1,
+    failed: 1,
+    skipped: 0,
+    total: 2
+  },
+  runtime: 3
+}
+```
+
+## Usage
+
+Listen to the events and receive the emitted data:
+
+```js
+var runner;
+
+// Attach the adapter.
+if (QUnit) {
+  runner = new JsReporters.QUnitAdapter(QUnit);
+} else if (mocha) {
+  runner = new JsReporters.MochaAdapter(mocha);
+ } else if (jasmine) {
+  runner = new JsReporters.JasmineAdapter(jasmine.getEnv());
+ }
+ 
+ // Listen to the same events for any testing framework.
+ runner.on('runStart', function(globalSuite) {
+   ...
+ });
+ 
+ runner.on('testEnd', function(test) {
+   ...
+ });
+ 
+ // Or use one of the built-in reporters.
+ JsReporters.TapReporter.init(runner);
+```
+
+## Integrations
+
+ - [BrowserStack](https://github.com/browserstack/browserstack-runner/blob/master/lib/_patch/reporter.js) 
```

Comment:
Its "browserstack-runner", BrowserStack is the product, browserstack-runner the tool making use of the product.